consensus/cometbft/service/finalize_block.go

```diff
@@ -34,73 +37,99 @@ func (s *Service) finalizeBlock(
 	ctx context.Context,
 	req *cmtabci.FinalizeBlockRequest,
 ) (*cmtabci.FinalizeBlockResponse, error) {
-	res, err := s.finalizeBlockInternal(ctx, req)
-	if res != nil {
-		res.AppHash = s.workingHash()
```

Comment:
just moved this to `formatFinalizeBlockResponse` method and drop the need of a `finalizeBlockInternal`